### PR TITLE
lwip: port to v2.0.2

### PR DESCRIPTION
**Pull request by miri64**
Finally came around to do that.

`tests/lwip_sock_ip` hangs with `LWIP_IPV4=1` and `LWIP_IPV6=1`, but there is strong evidence that this is related to #7035.

### CODE REVIEW

File: pkg/lwip/contrib/sys_arch.c [hunk 1]
```diff
@@ -36,6 +36,11 @@ void sys_init(void)
     return;
 }
 
+u32_t sys_now(void)
+{
+    return (uint32_t)(xtimer_now_usec64() / 1000);
```
Comment (smlng) [inline pkg/lwip/contrib/sys_arch.c]: s/1000/US_PER_MS/ ?

Comment (miri64) [inline pkg/lwip/contrib/sys_arch.c]: Addressed

**Comment by smlng** [comment]
for completeness: same compile error is also reported for `htonl`

**Comment by miri64** [comment]
for the current one?

**Comment by smlng** [comment]
> for the current one?

what do you mean? LWIP version in master compiles on macOS, LWIP version of this PR, does not.

Compile error:
```
In file included from /Volumes/workspace/github/smlng/RIOT/tests/lwip/bin/pkg/native/lwip/src/include/lwip/ip_addr.h:41:
/Volumes/workspace/github/smlng/RIOT/tests/lwip/bin/pkg/native/lwip/src/include/lwip/def.h:96:9: error: 'htons' macro redefined [-Werror,-Wmacro-redefined]
#define htons(x) lwip_htons(x)
        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk/usr/include/sys/_endian.h:133:9: note: previous definition is here
#define htons(x)        __DARWIN_OSSwapInt16(x)
        ^
In file included from /Volumes/workspace/github/smlng/RIOT/pkg/lwip/contrib/sock/ip/lwip_sock_ip.c:21:
In file included from /Volumes/workspace/github/smlng/RIOT/sys/include/net/sock/ip.h:451:
In file included from /Volumes/workspace/github/smlng/RIOT/pkg/lwip/include/sock_types.h:22:
In file included from /Volumes/workspace/github/smlng/RIOT/tests/lwip/bin/pkg/native/lwip/src/include/lwip/api.h:47:
In file included from /Volumes/workspace/github/smlng/RIOT/tests/lwip/bin/pkg/native/lwip/src/include/lwip/netbuf.h:47:
In file included from /Volumes/workspace/github/smlng/RIOT/tests/lwip/bin/pkg/native/lwip/src/include/lwip/ip_addr.h:41:
/Volumes/workspace/github/smlng/RIOT/tests/lwip/bin/pkg/native/lwip/src/include/lwip/def.h:97:9: error: 'ntohs' macro redefined [-Werror,-Wmacro-redefined]
#define ntohs(x) lwip_ntohs(x)
        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk/usr/include/sys/_endian.h:132:9: note: previous definition is here
#define ntohs(x)        __DARWIN_OSSwapInt16(x)
        ^
In file included from /Volumes/workspace/github/smlng/RIOT/pkg/lwip/contrib/sock/ip/lwip_sock_ip.c:21:
In file included from /Volumes/workspace/github/smlng/RIOT/sys/include/net/sock/ip.h:451:
In file included from /Volumes/workspace/github/smlng/RIOT/pkg/lwip/include/sock_types.h:22:
In file included from /Volumes/workspace/github/smlng/RIOT/tests/lwip/bin/pkg/native/lwip/src/include/lwip/api.h:47:
In file included from /Volumes/workspace/github/smlng/RIOT/tests/lwip/bin/pkg/native/lwip/src/include/lwip/netbuf.h:47:
In file included from /Volumes/workspace/github/smlng/RIOT/tests/lwip/bin/pkg/native/lwip/src/include/lwip/ip_addr.h:41:
/Volumes/workspace/github/smlng/RIOT/tests/lwip/bin/pkg/native/lwip/src/include/lwip/def.h:98:9: error: 'htonl' macro redefined [-Werror,-Wmacro-redefined]
#define htonl(x) lwip_htonl(x)
        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk/usr/include/sys/_endian.h:136:9: note: previous definition is here
#define htonl(x)        __DARWIN_OSSwapInt32(x)
        ^
In file included from /Volumes/workspace/github/smlng/RIOT/pkg/lwip/contrib/sock/ip/lwip_sock_ip.c:21:
In file included from /Volumes/workspace/github/smlng/RIOT/sys/include/net/sock/ip.h:451:
In file included from /Volumes/workspace/github/smlng/RIOT/pkg/lwip/include/sock_types.h:22:
In file included from /Volumes/workspace/github/smlng/RIOT/tests/lwip/bin/pkg/native/lwip/src/include/lwip/api.h:47:
In file included from /Volumes/workspace/github/smlng/RIOT/tests/lwip/bin/pkg/native/lwip/src/include/lwip/netbuf.h:47:
In file included from /Volumes/workspace/github/smlng/RIOT/tests/lwip/bin/pkg/native/lwip/src/include/lwip/ip_addr.h:41:
/Volumes/workspace/github/smlng/RIOT/tests/lwip/bin/pkg/native/lwip/src/include/lwip/def.h:99:9: error: 'ntohl' macro redefined [-Werror,-Wmacro-redefined]
#define ntohl(x) lwip_ntohl(x)
        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk/usr/include/sys/_endian.h:135:9: note: previous definition is here
#define ntohl(x)        __DARWIN_OSSwapInt32(x)
        ^
4 errors generated.
make[2]: *** [/Volumes/workspace/github/smlng/RIOT/tests/lwip/bin/native/lwip_sock_ip/lwip_sock_ip.o] Error 1
make[1]: *** [ALL--/Volumes/workspace/github/smlng/RIOT/pkg/lwip/contrib/sock/ip] Error 2
```

**Comment by miri64** [comment]
>> for the current one?
>
> what do you mean?

I meant the `htonl` in RIOT's `core` ;-)

**Comment by miri64** [comment]
@smlng can you check with the latest commit?

**Comment by smlng** [comment]
sure, but last 2 commits didn't help -> same error

What helps is: add to `Makefile.include` (before all other module filters):

```
ifeq ($(TOOLCHAIN), llvm)
    CFLAGS += -Wno-macro-redefined
endif
```

**Comment by smlng** [comment]
but not sure if that's the way to go

**Comment by miri64** [comment]
Okay let's try it via lwIP's options

**Comment by smlng** [comment]
looks good now on macOS, too. Please squash

**Comment by miri64** [comment]
Squashed

**Comment by smlng** [comment]
ACK, and GO when Murdock is happy, too ...

**Pull request merged**